File: unit_3/ruby/week_7/2-PezDispenser/my_solution.rb
# PezDispenser Class from User Stories

# Review the following user stories:
# - As a pez user, I'd like to be able to "create" a new pez dispenser with a group of flavors that
#      represent pez so it's easy to start with a full pez dispenser.
# - As a pez user, I'd like to be able to easily count the number of pez remaining in a dispenser
#      so I can know how many are left.
# - As a pez user, I'd like to be able to take a pez from the dispenser so I can eat it.
# - As a pez user, I'd like to be able to add a pez to the dispenser so I can save a flavor for later.
# - As a pez user, I'd like to be able to see all the flavors inside the dispenser so I know the order
#      of the flavors coming up.

class PezDispenser

# your code here!

end

# DRIVER CODE 

flavors = %w(cherry chocolate cola grape lemon orange peppermint raspberry strawberry).shuffle
super_mario = PezDispenser.new(flavors)
puts "A new pez dispenser has been created. You have #{super_mario.pez_count} pez!"
puts "Here's a look inside the dispenser:"
puts super_mario.see_all_pez
puts "Adding a purple pez."
super_mario.add_pez("purple")   # mmmmm, purple flavor
puts "Now you have #{super_mario.pez_count} pez!"
puts "Oh, you want one do you?"
puts "The pez flavor you got is: #{super_mario.get_pez}"
puts "Now you have #{super_mario.pez_count} pez!"

# WRITE TESTS BELOW THIS LINE

File: unit_3/cs/week_7/3-PezDispenser-solo-challenge/my_solution.cs
// Pseudo-code

// Initial solution
using System;
using System.Collections;

class PezDispenser {
	public static void Main() {
		String[] flavors = new String[] { "cola", "chocolate", "strawberry", "raspberry", "grape", "orange", "cherry", "peppermint", "lemon" };
		PezDispenser super_mario = new PezDispenser(flavors);
		Console.WriteLine( "A new pez dispenser has been created. You have " + super_mario.PezCount + " pez!");
		Console.WriteLine( "Here's a look inside the dispenser:" + super_mario.SeeAllPez() );
		Console.WriteLine( "Adding a purple pez." );
		super_mario.AddPez("purple");   // mmmmm, purple flavor
		Console.WriteLine( "Now you have " + super_mario.PezCount + " pez!" );
		Console.WriteLine( "Oh, you want one do you?" );
		Console.WriteLine( "The pez flavor you got is: " + super_mario.GetPez() );
		Console.WriteLine( "Now you have " + super_mario.PezCount + " pez!" );
	}
}

// Refactored solution

// Reflections
